### PR TITLE
debug: check that reading nonexistent CSR returns error.

### PR DESCRIPTION
**Pull request by mwachs5**
This isn't ready to merge, as it fails, and OpenOCD needs to be bumped.

@timsifive can you take a look? If I run this test with my new version of OpenOCD which returns errors, it does correctly print the expected string in the test log, but then the overall test fails. I think I'm not doing the actual `assertIn` correctly.

### CODE REVIEW

File: debug/gdbserver.py [hunk 1]
```diff
@@ -567,6 +567,13 @@ def test(self):
         assertEqual(123, self.gdb.p("$x1"))
         assertEqual(123, self.gdb.p("$csr832"))
 
+class NonExistentCsr(RegsTest):
+    def test(self):
+        #Print old version of $misa at 0xF10 (moved to 0x301)
+        self.gdb.p("$csr3856")
```
Comment (timsifive) [inline debug/gdbserver.py]: output = ...

Comment (mwachs5) [inline debug/gdbserver.py]: Sorry, not getting it. I tried 
`output = self.gdb.p("..")` 
but that's wrong.  Maybe it's on stderr or something:

```
(gdb) p/x $csr3856
p/x $csr3856
Could not fetch register "csr3856"; remote failure reply 'E0E'
(gdb) ////////////////////////////////////////
::::::::[ ran 2 tests in 124s ]:::::::::
1 tests returned exception
    NonExistentCsr
1 tests returned pass
```

Comment (mwachs5) [inline debug/gdbserver.py]: NM I read testlib.py and get why it doesn't work. Will figure it out.

Comment (timsifive) [inline debug/gdbserver.py]: Your test returned an exception. There should be more output that details exactly what happened, but I suspect you hit this code in p():
```
    def p(self, obj):
        output = self.command("p/x %s" % obj)
        m = re.search("Cannot access memory at address (0x[0-9a-f]+)", output)
        if m:
            raise CannotAccess(int(m.group(1), 0))
        value = int(output.split('=')[-1].strip(), 0)
        return value
```

So your tests needs to:
```
try:
   self.gdb.p(...)
   assert False, "access should have failed"
except CannotAccess:
   # expected
   pass
```

just like MemTestReadInvalid

**Comment by mwachs5** [comment]
Ok, this works now for this test, but for the 64-bit tests I now get an error about -march needing to be set to use 'D' instructions. I hadn't bumped my toolchain in a while, does something need to change in the way I'm invoking the test?

**Comment by mwachs5** [comment]
Actually this doesn't work against Spike  yet, still digging in.

**Comment by mwachs5** [comment]
It works in Spike if I just run Spike and GDB, but I get a timeout if I try to run it through GDB Server...

**Comment by mwachs5** [comment]
@timsifive what does it mean to "pass" these tests? Can I just run make debug-check? Or something else?

**Comment by timsifive** [comment]
`make debug-check` should do it. What I do is I cd into the debug
directory, and run `make clean && make all`. This runs tests against both
32- and 64-bit spike. Testing against hardware you have to manually by
running gdbserver.py directly.

**Pull request closed without merging**